src/providers/trusted_service/context/generate_random.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::error::Error;
+use super::ts_protobuf::{GenerateRandomIn, GenerateRandomOut};
+use super::Context;
+use log::info;
+use std::convert::TryInto;
+
+impl Context {
+    pub fn generate_random(&self, size: usize) -> Result<Vec<u8>, Error> {
+        info!("Handling GenerateRandom request");
+        let open_req = GenerateRandomIn {
+            size: size.try_into().unwrap(),
+        };
+        match self.send_request(&open_req) {
+            Ok(GenerateRandomOut { random_bytes }) => Ok(random_bytes),
+            Err(e) => Err(e),
+        }
```

Comment:
I think you could simplify this to something like:
```Rust
Ok(self.send_request(&open_req)?.random_bytes)
```
But this is just a bit of bikeshedding